File: L274-h_index.py
# https://leetcode.com/problems/h-index/
class Solution:
    def hIndex(self, citations: List[int]) -> int:
        maxi= max(citations)
        length = len(citations)
        p = length - 1
        counts = [0 for _ in range(maxi + 1)]
        for citation in citations:
            counts[citation] += 1
        for i in range(maxi + 1):
            for count in range(counts[i]):
                citations[p] = i
                p -= 1
        for i in range(length - 1, -1, -1):
            if (citations[i] >= i + 1):
                return i + 1
        return citations[0]

File: Top_Interview_150_List_With_solution(leetcode)/Array|String/L274-H-Index.java
// https://leetcode.com/problems/h-index/

class Solution {
    public int hIndex(int[] citations) {
        int max = 0, p = citations.length - 1, length = citations.length;
        for (int citation: citations) {
            max = Math.max(max, citation);
        }
        int counts[] = new int[max + 1];
        for (int citation: citations) {
            counts[citation]  += 1;
        }
        for (int i = 0; i <= max; i++) {
            for (int j = 0; j < counts[i]; j++) {
                citations[p--] = i;
            }
        }
        for (int i = length - 1; i >= 0; i--) {
            if (citations[i] >= i + 1) {
                return i + 1;
            }
        }
        return citations[0];
    }
}
